snapcraft/internal/cache/_snap.py

```diff
@@ -45,7 +45,10 @@ def cache(self, snap_filename, revision):
             revision)
         cached_snap_path = os.path.join(self.snap_cache_dir, cached_snap)
         try:
-            link_or_copy(snap_filename, cached_snap_path)
+            if not os.path.isfile(cached_snap_path):
+                # this must not be hard-linked,
+                # hence avoids using fileutils.link_or_copy
+                shutil.copyfile(snap_filename, cached_snap_path)
```

Comment:
(Note that I understand why given the PR description, but someone looking at this won't)